### PR TITLE
`All` pipeline improvements

### LINKED ISSUE #117: Clean up intermediary load job tables on BQ
I wrote a wildcard query over the `technologies` dataset and found ~hundreds of tables with names like `httparchive.technologies.beam_bq_job_LOAD_httparchivejulystreamingpipeline_LOAD_STEP_126_013fc2c379fe0768444c5cf87ef6c4c3_005eb94967ab4afa85b3ad236fcbd859`. It seems like the `httparchivejulystreamingpipeline` Dataflow job failed while writing data into BQ and these intermediary tables were never cleaned up.

image

Any table prefixed with `beam` is an intermediary table and can safely be deleted. We can write a DML query to target and delete these. We should repeat this process for the other datasets that Dataflow writes into.

### PR DESCRIPTION
**Pull request by giancarloaf**
**Changes**
* Readme updates
    * add table of contents
    * add instructions to setup and use manifest files in the streaming pipeline
    * add instructions to delete temp tables left behind by jobs using BigQuery `FILE_LOADS` (Fixes #117)
    * add known issue for "Error 413 (Request Entity Too Large)!!1"
* add functionality to run streaming pipeline with a backfill pub/sub topic (for manifest files)
* refactoring to share pipeline input parameters across `combined` and `all` pipelines
* add additional constants for BigQuery table creation to include partitioning and clustering
* add unit tests for all pipeline serialization

**Validation**

small job: https://console.cloud.google.com/dataflow/jobs/us-west1/2022-07-03_00_19_12-12992533282090519155;mainTab=JOB_GRAPH?project=httparchive

large job (Jun_9_2022): https://console.cloud.google.com/dataflow/jobs/us-west1/2022-07-05_09_40_00-14123137719167014898?project=httparchive

large job (android-Jul_1_2022): https://console.cloud.google.com/dataflow/jobs/us-west1/2022-07-09_19_24_44-928549535619831989?project=httparchive

large job (chrome-Jul_1_2022): https://console.cloud.google.com/dataflow/jobs/us-west1/2022-07-10_16_23_40-2491574173175659024?project=httparchive

~~streaming job using manifest file as input (Jul_1_2022): https://console.cloud.google.com/dataflow/jobs/us-west1/2022-07-19_22_14_11-10074311526941494587?project=httparchive~~  **stopped testing streaming, will be deprecated in a future change**

### CODE REVIEW

**Comment by rviscomi** [comment]
Let me know when this is ready for review! 👍

File: modules/constants.py [hunk 1]
```diff
@@ -38,6 +40,16 @@ def _get_schema(path):
...
+    "additional_bq_parameters": {
+        "all_pages": {
+            'timePartitioning': {'type': 'DAY', 'field': 'date', 'requirePartitionFilter': True},
+            'clustering': {'fields': ['client', 'is_root_page', 'rank']}
+        },
+        "all_requests": {
+            'timePartitioning': {'type': 'DAY', 'field': 'date', 'requirePartitionFilter': True},
+            'clustering': {'fields': ['client', 'is_root_page', 'is_main_document', 'type']}
+        },
+    },
```
Comment (giancarloaf) [inline modules/constants.py:52]: note: this is added to aid with table creation. I used the existing table configuration as a reference, but the BigQuery API can be referenced for more
https://cloud.google.com/bigquery/docs/reference/rest/v2/Job#jobconfigurationload

File: modules/import_all.py [hunk 2]
```diff
@@ -3,134 +3,155 @@
...
+    HarJsonToSummary,
+)
 
-    hashed_url = hash_url(page_url)
-    if hashed_url % NUM_PARTITIONS != index:
-        return
+# BigQuery can handle rows up to 100 MB when using `WriteToBigQuery.Method.FILE_LOADS`
+MAX_CONTENT_SIZE_100_MB = 10 * 1024 * 1024
+# BigQuery can handle rows up to 100 MB when using `WriteToBigQuery.Method.STREAMING_INSERTS`
+MAX_CONTENT_SIZE_10_MB = 1024 * 1024
```
Comment (rviscomi) [inline modules/import_all.py:27]: Should these be enums?

I'm also confused why they're named 100 when the value is 10, and 10 when the value is 1.

**Comment by giancarloaf** [comment]
Notes from offline sync
- Remove the size logging functionality, it is not actionable. The size issues we encountered in previous runs were out of our control due to the way the Beam SDK handles batching.
- Add a note in the readme to document how the size limits we encountered were discovered.

File: modules/import_all.py [hunk 3]
```diff
@@ -435,99 +469,121 @@ def to_json(obj):
...
+        partial(get_page, max_content_size)
     )
-    known_args, pipeline_args = parser.parse_known_args(argv)
-    pipeline_options = PipelineOptions(pipeline_args, save_main_session=True)
 
-    pipeline = beam.Pipeline(options=pipeline_options)
+    if logging.getLogger().isEnabledFor(logging.DEBUG):
+        mapped_pages = mapped_pages | "LogPagesSize" >> beam.Map(
+            partial(log_size, "page", max_content_size)
+        )
```
[the remark this answers is not in the document]
Comment (giancarloaf) [reply modules/import_all.py]: killed this completely, not needed.

**Review by rviscomi — APPROVED** [review]
LGTM!

We'll resolve the rest of the FRs in a follow-up PR

**Pull request merged**